### PR TITLE
chore(release): update package versions

### PR DESCRIPTION
**Pull request by abbeyhrt**
Changes:
 - @carbon/cli: 10.5.0 => 10.6.0-rc.0
 - @carbon/colors: 10.5.0 => 10.6.0-rc.0
 - carbon-components: 10.7.0 => 10.8.0-rc.0
 - @carbon/elements: 10.7.0 => 10.8.0-rc.0
 - @carbon/grid: 10.6.0 => 10.7.0-rc.0
 - @carbon/icons-handlebars: 10.6.0 => 10.7.0-rc.0
 - @carbon/icons-react: 10.6.0 => 10.7.0-rc.0
 - @carbon/icons-vue: 10.6.0 => 10.7.0-rc.0
 - @carbon/icons: 10.6.0 => 10.7.0-rc.0
 - @carbon/layout: 10.5.0 => 10.6.0-rc.0
 - carbon-components-react: 7.7.0 => 7.8.0-rc.0
 - @carbon/themes: 10.7.0 => 10.8.0-rc.0
 - @carbon/type: 10.6.0 => 10.7.0-rc.0

Reviewers:

Double check that packages were updated to the correct versions
Double check that all tests are passing ✅
Double check that everything is working as expected

### CODE REVIEW

**Review by alisonjoseph — APPROVED** [review]

**Comment by alisonjoseph** [comment]
LGTM once conflict is fixed

File: packages/themes/package.json [hunk 1]
```diff
@@ -24,14 +24,13 @@
     "build": "yarn clean && bundler bundle src/index.js --name CarbonThemes && node tasks/build.js && bundler check \"scss/*.scss\" && bundler sassdoc \"scss/**/*.scss\""
   },
   "dependencies": {
-    "@carbon/colors": "10.5.0",
-    "@carbon/type": "10.6.0",
-    "@carbon/layout": "10.5.0"
```
Comment (vpicone) [inline packages/themes/package.json:29]: Did we loose a dependency here?

**Review by aledavila — APPROVED** [review]

**Review by jillianhowarth — APPROVED** [review]

File: packages/themes/package.json (hunk 1 above)
Comment (abbeyhrt) [reply packages/themes/package.json:29]: Yeah, I don't know why. I'll add it back in!

**Review by joshblack — APPROVED** [review]

**Comment by joshblack** [comment]
@abbeyhrt gonna close this to pursue 10.9 like we were talking about!

**Pull request closed without merging**